src/app/Scenes/MyCollection/Screens/ArtworkForm/MyCollectionArtworkForm.tsx

```diff
@@ -260,7 +262,7 @@ export const MyCollectionArtworkForm: React.FC<MyCollectionArtworkFormProps> = (
           <SavingArtworkModal
             testID="saving-artwork-modal"
             isVisible={loading}
-            loadingText={isArtworkSaved ? "Generating market data" : "Saving artwork"}
+            loadingText={isArtworkSaved ? displayText : "Saving artwork"}
```

Comment:
We could initialize  `displayText` with "Saving artwork" and then just pass:

```suggestion
            loadingText={displayText}
```

Reply:
We need to check hasMarketPriceInsights, do that in the promise; and at some point set isArtworkSaved to true or false. So I would keep it as it is

Reply:
Not saying the approach doesn't work, but I think we can simplify the logic like this and avoid the duplication of the string.

Reply:
I guess not
I can do that if it looks better to you 
const [savingArtworkModalDisplayText, setSavingArtworkModalDisplayText] = useState<string>("Saving artwork")
if (!!hasMarketPriceInsights) { setSavingArtworkModalDisplayText("Generating market data")}